### ISSUE TITLE
[Suggestion] Channels subscriptions "mix"

### ISSUE THREAD

**Issue by greenfoo** [lekma/plugin.video.invidious#21]
Would it be possible to have something like this?

1. I visit some channels and "save" ("subscribe") them
2. A new folder appears which contains the latest videos from all those channels sorted by "newest".

Example: I subscribe to channels A, B and C. This new folder (maybe called "Subscriptions RSS"?) would then contain something like this (in this order):
* Video from today from Channel A
* Video from yesterday from Channel A
* Video from last week from Chanel B
* Video from last month from Channel A
* Video from last year from Channel C

This would allow us to have the "home page" functionality of a YouTube account without actually needing one.

Thanks.

**Comment by lekma** [comment]
disclaimer: I don't have a youtube account (never had one) so I don't really get what you mean with:

> This would allow us to have the "home page" functionality of a YouTube account without actually needing one.

what we could maybe do is: get the latest videos from all the channels you 'follow' and try to sort them by their published date (knowing that the dates from invidious are quite borked right now, see https://github.com/omarroth/invidious/issues/1184).

I think we'd have to limit the number of videos we extract from each channel (to avoid long folder loading time), how many videos would you like to see from each of these authors?

also, how would you call the main folder? following? feed? (i think we should avoid the word 'subscribe' and/or 'subscriptions' for being to close to what youtube uses).

maybe I misunderstood what you meant, if that is the case, please, feel free to correct me.

**Comment by greenfoo** [comment]
> what we could maybe do is: get the latest videos from all the channels you 'follow' and try to sort them by their published date

Yes, sorry, that's exactly what I was (poorly) trying to explain :)

I must confess I don't have a YouTube account either. I was taking "inspiration" from FreeTube interface (also based on Invidious) which looks [like this](https://freetubeapp.io/images/FreeTube1.png).

Notice how on the "main" view (which they call "Subscriptions" and uses the RSS logo) they list "_the latest videos from all the channels you follow sorted by published date_" (just as you suggested).

I guess 5 videos from each channel should be more than enough, but maybe you could also make it a user defined setting, in case someone wants to tune it (?)

I agree that "Subscription" is not a good word for this. "Feed", as you suggested, sounds much better.

Thanks!

**Comment by lekma** [comment]
there are probably still a couple of things needed, but i'd be grateful if you had the time to test a2d6854371a51b8100727485bfdbf973b8279e92 and report any bug/inconsistency you might find.

**Comment by greenfoo** [comment]
I just tested it and this is exactly what I had in mind! Thanks.

It took me a few seconds to figure out how to remove a channel from the Feed, but now that I know how, it makes perfect sense.

Two possible improvements for the future that I can think of:
1. At the end of the feed, and a "Load more..." entry that loads the next block of "X" videos from each channel. There is one problem with this, thought: one or more of the new loaded videos *might* have a later "published" date than one or more of the already listed entries. I think that the easiest solution to this problem is to show the next batch of results on its own (new) page instead of appending entries to the already existing list. What do you think?
2. It would be nice to be able to create "custom feeds". So, by default, only one "Feed" folder exists (as it is the case now), but in the "Settings" menu there is an option to rename the already existing "Feed" folder and create new ones. When you right click on a "channel" you get to select to which feed you want to add it to (maybe more than one, so a multi selection box would be needed). To distinguish "feed" folders from the rest, their name would always have some prefix (maybe a [unicode symbol](https://unicode-table.com/es/#058E)?)

Also, one last suggestion: would it be possible to start loading the contents of the different folders in the background instead of having to wait for the user to actually click on the feed folder? (I'm not even sure this is possible using Kodi's API).

Anyway, these are all just suggestions. The plugin as it is now is perfect for all I want.

Thanks!

**Comment by lekma** [comment]
> At the end of the feed, and a "Load more..." entry that loads the next block of "X" videos from each channel. There is one problem with this, thought: one or more of the new loaded videos might have a later "published" date than one or more of the already listed entries. I think that the easiest solution to this problem is to show the next batch of results on its own (new) page instead of appending entries to the already existing list. What do you think?

I implemented a version of that in 821bdd44cee4eecddfea5f74c99c732d9dbf3881, I hope it kinda fits your expectations.

> It would be nice to be able to create "custom feeds". So, by default, only one "Feed" folder exists (as it is the case now), but in the "Settings" menu there is an option to rename the already existing "Feed" folder and create new ones. When you right click on a "channel" you get to select to which feed you want to add it to (maybe more than one, so a multi selection box would be needed). To distinguish "feed" folders from the rest, their name would always have some prefix (maybe a unicode symbol?)

This will not happen - let me rephrase that - I will not make it happen. If you (or someone else) want to submit a patch, then yeah, I'll have a look at it. That suggestion is overly complicated and I think you're better off building a favourites addon that handles all your needs. The simple 'Feed' is as far as I will go :smiley: (also the context menu on a video feels crowded right now).

> Also, one last suggestion: would it be possible to start loading the contents of the different folders in the background instead of having to wait for the user to actually click on the feed folder? (I'm not even sure this is possible using Kodi's API).

I might have a look at that later (maybe for a next major release). With the current latest modifications the full feed is loaded the first time you enter the folder and then only if you re-enter the folder and 10 mins have elapsed or you modifed the feed's channels.

Would you mind testing 821bdd44cee4eecddfea5f74c99c732d9dbf3881 and report back?

thanks for your help

**Comment by greenfoo** [comment]
It works great!

Removing the "Videos per channel" option I also think is a good idea (as it was kind of confusing to explain that it only applied to feeds).

And the 10 minutes cache makes it now much snappier to navigate.

Thanks again.

**Comment by lekma** [comment]
ok, thanks for the feedback.
